blocks/algorithms/__init__.py

```diff
@@ -590,3 +590,29 @@ def compute_step(self, param, previous_step):
                    (time, t1)]
 
         return step, updates
+
+
+class RemoveNotFinite(StepRule):
+    """A step rule that replaces non-finite elements.
+
+    Replaces non-finite elements (`inf` or `NaN`) in the step with
```

Comment:
Yeah, this is not correct. @dwf suggested this docstring. (For some odd reason, his comment was on my fork not here.)
How about replacing first line with:
``Replaces steps with non-finite norm (`inf` or `NaN`) with``